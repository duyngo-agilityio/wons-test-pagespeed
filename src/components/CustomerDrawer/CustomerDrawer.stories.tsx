import type { Meta, StoryObj } from '@storybook/react';

// Components
import { CustomerDrawer } from '@/components';

const meta: Meta<typeof CustomerDrawer> = {
  title: 'Components/CustomerDrawer',
  component: CustomerDrawer,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;
type Story = StoryObj<typeof CustomerDrawer>;

export const Default: Story = {
  args: {},
};
