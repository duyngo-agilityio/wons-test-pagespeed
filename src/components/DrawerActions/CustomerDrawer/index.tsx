'use client';

import Drawer from 'react-modern-drawer';
import 'react-modern-drawer/dist/index.css';

// icons
import { IoClose } from 'react-icons/io5';

// components
import { CustomerForm, Button } from '@/components';

interface CustomerDrawerProps {
  isDrawerOpen: boolean;
  toggleDrawer: () => void;
}

const CustomerDrawer = ({
  isDrawerOpen,
  toggleDrawer,
}: CustomerDrawerProps): JSX.Element => {
  const handleFormSubmit = () => {
    // TODO: later
  };

  return (
    <Drawer
      open={isDrawerOpen}
      onClose={toggleDrawer}
      direction="right"
      size={400}
      className="!w-full md:!w-[400px]"
    >
      <div className="p-5 relative bg-white dark:bg-gray-400 h-full max-w-full">
        <Button
          onClick={toggleDrawer}
          className="absolute top-5 right-5 !bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
        >
          <IoClose size={20} />
        </Button>
        <CustomerForm onSubmit={handleFormSubmit} />
      </div>
    </Drawer>
  );
};

export default CustomerDrawer;
