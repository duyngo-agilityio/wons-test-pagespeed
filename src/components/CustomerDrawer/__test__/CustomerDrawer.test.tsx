import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import CustomerDrawer from '..';

describe('CustomerDrawer', () => {
  it('matches snapshot', () => {
    const { container } = render(<CustomerDrawer />);
    expect(container).toMatchSnapshot();
  });

  it('opens the drawer when the button is clicked', () => {
    const { getAllByRole, getByRole } = render(<CustomerDrawer />);
    const buttons = getAllByRole('button', { name: 'Add Customer' });
    const button = buttons[0];
    fireEvent.click(button);
    expect(getByRole('heading', { name: 'Add Customer' })).toBeInTheDocument();
  });

  it.skip('closes the drawer when the close button is clicked', () => {
    const { getAllByRole, getByTestId, getByRole } = render(<CustomerDrawer />);
    const buttons = getAllByRole('button', { name: 'Add Customer' });
    const addButton = buttons[0];
    fireEvent.click(addButton);

    const closeButton = getByTestId('close-button');
    fireEvent.click(closeButton);

    const navigation = getByRole('navigation');
    expect(navigation).toHaveStyle({ transform: 'translate3d(100%, 0, 0)' });
  });
});
