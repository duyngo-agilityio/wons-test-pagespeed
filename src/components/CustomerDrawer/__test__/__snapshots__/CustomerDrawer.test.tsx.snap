// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDrawer matches snapshot 1`] = `
<div>
  <div
    class="w-full flex justify-center md:justify-end"
  >
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 pb-2.5 pt-[11px] rounded-[10px] border-[1px] min-w-max text-xl font-medium md:w-auto h-10 px-2.5 w-full mt-10 md:mt-0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="text-white"
        fill="currentColor"
        focusable="false"
        height="22"
        stroke="currentColor"
        stroke-width="0"
        tabindex="-1"
        viewBox="0 0 16 16"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
        />
      </svg>
      <div
        class="flex items-center"
      >
        Add Customer
      </div>
    </button>
    <div
      class="EZDrawer"
      id="EZDrawer"
    >
      <input
        class="EZDrawer__checkbox"
        id="EZDrawer__checkbox"
        type="checkbox"
      />
      <nav
        class="EZDrawer__container !w-full md:!w-[450px] max-h-screen"
        id="EZDrawer__container"
        role="navigation"
        style="z-index: 101; transition-duration: 500ms; top: 0px; right: 0px; transform: translate3d(100%, 0, 0); width: 400px; height: 100vh;"
      >
        <div
          class="p-8 bg-white dark:bg-gray-400 h-full max-w-full overflow-y-auto"
        >
          <button
            class="z-0 group box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover border-transparent hover:!bg-gray-200/50 dark:hover:!bg-gray-900 px-5 pb-2.5 pt-[11px] text-md border-[1px] min-w-max absolute top-5 right-5 !bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
            data-testid="close-button"
            type="button"
          >
            <div
              class="flex items-center"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
                />
              </svg>
            </div>
          </button>
          <form
            class="w-full max-w-2xl mx-auto"
          >
            <h2
              aria-level="2"
              class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl "
            >
              Add Customer
            </h2>
            <div
              class="flex justify-center mt-[21px]"
            >
              <div
                class="flex flex-col justify-center items-center my-20"
              >
                <label
                  class="cursor-pointer hover:scale-110 transition-transform"
                  for="file"
                >
                  <div
                    class="rounded-full w-32 h-32 bg-gray-50 dark:bg-gray-600 flex justify-center items-center"
                  >
                    <svg
                      class="text-blue-800/70 dark:text-white/70"
                      fill="currentColor"
                      height="32"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="256"
                        cy="272"
                        r="64"
                      />
                      <path
                        d="M432 144h-59c-3 0-6.72-1.94-9.62-5l-25.94-40.94a15.52 15.52 0 0 0-1.37-1.85C327.11 85.76 315 80 302 80h-92c-13 0-25.11 5.76-34.07 16.21a15.52 15.52 0 0 0-1.37 1.85l-25.94 41c-2.22 2.42-5.34 5-8.62 5v-8a16 16 0 0 0-16-16h-24a16 16 0 0 0-16 16v8h-4a48.05 48.05 0 0 0-48 48V384a48.05 48.05 0 0 0 48 48h352a48.05 48.05 0 0 0 48-48V192a48.05 48.05 0 0 0-48-48zM256 368a96 96 0 1 1 96-96 96.11 96.11 0 0 1-96 96z"
                      />
                    </svg>
                  </div>
                </label>
                <div
                  class="group flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] hidden"
                  data-filled="true"
                  data-filled-within="true"
                  data-invalid="true"
                  data-slot="base"
                >
                  <div
                    class="h-full flex flex-col"
                    data-slot="main-wrapper"
                  >
                    <div
                      class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background !bg-danger-50 data-[hover=true]:!bg-danger-100 bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
                      data-slot="input-wrapper"
                      style="cursor: text;"
                    >
                      <div
                        class="inline-flex w-full items-center h-full box-border"
                        data-slot="inner-wrapper"
                      >
                        <input
                          accept="image/*"
                          aria-invalid="true"
                          aria-label="Upload Avatar"
                          class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !placeholder:text-danger !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                          data-slot="input"
                          id="file"
                          placeholder=" "
                          type="file"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] h-[71px]"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px]"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <label
                    class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                    data-slot="label"
                    for="react-aria-:r5:"
                    id="react-aria-:r6:"
                  >
                    First Name
                  </label>
                  <div
                    class="inline-flex w-full items-center h-full box-border"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="First Name"
                      aria-labelledby="react-aria-:r5: react-aria-:r6:"
                      class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                      data-slot="input"
                      id="react-aria-:r5:"
                      placeholder=" "
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] h-[71px]"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px]"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <label
                    class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                    data-slot="label"
                    for="react-aria-:ra:"
                    id="react-aria-:rb:"
                  >
                    Last Name
                  </label>
                  <div
                    class="inline-flex w-full items-center h-full box-border"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Last Name"
                      aria-labelledby="react-aria-:ra: react-aria-:rb:"
                      class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                      data-slot="input"
                      id="react-aria-:ra:"
                      placeholder=" "
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] h-[71px]"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px]"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <label
                    class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                    data-slot="label"
                    for="react-aria-:rf:"
                    id="react-aria-:rg:"
                  >
                    Email Address
                  </label>
                  <div
                    class="inline-flex w-full items-center h-full box-border"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Email Address"
                      aria-labelledby="react-aria-:rf: react-aria-:rg:"
                      class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                      data-slot="input"
                      id="react-aria-:rf:"
                      placeholder=" "
                      type="email"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] mb-12"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <label
                    class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                    data-slot="label"
                    for="react-aria-:rk:"
                    id="react-aria-:rl:"
                  >
                    Phone Number
                  </label>
                  <div
                    class="inline-flex w-full items-center h-full box-border"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Phone Number"
                      aria-labelledby="react-aria-:rk: react-aria-:rl:"
                      class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                      data-slot="input"
                      id="react-aria-:rk:"
                      placeholder=" "
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] mb-12"
              data-filled="true"
              data-filled-within="true"
              data-has-elements="true"
              data-has-label="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <label
                    class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                    data-slot="label"
                    for="react-aria-:rp:"
                    id="react-aria-:rq:"
                  >
                    Job
                  </label>
                  <div
                    class="inline-flex w-full items-center h-full box-border"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Job"
                      aria-labelledby="react-aria-:rp: react-aria-:rq:"
                      class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                      data-slot="input"
                      id="react-aria-:rp:"
                      placeholder=" "
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="relative flex-1"
            >
              <div
                class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[29px] flex-1"
                data-filled="true"
                data-filled-within="true"
                data-has-elements="true"
                data-has-label="true"
                data-slot="base"
              >
                <div
                  class="h-full flex flex-col"
                  data-slot="main-wrapper"
                >
                  <div
                    class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
                    data-slot="input-wrapper"
                    style="cursor: text;"
                  >
                    <label
                      class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                      data-slot="label"
                      for="react-aria-:ru:"
                      id="react-aria-:rv:"
                    >
                      Address
                    </label>
                    <div
                      class="inline-flex w-full items-center h-full box-border"
                      data-slot="inner-wrapper"
                    >
                      <input
                        aria-label="Address"
                        aria-labelledby="react-aria-:ru: react-aria-:rv:"
                        class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                        data-has-end-content="true"
                        data-slot="input"
                        id="react-aria-:ru:"
                        placeholder=" "
                        type="text"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 20 20"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col w-full h-[71px] mb-5"
            >
              <div
                class="group flex flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] w-full border-gray-300 rounded-md"
                data-filled="true"
                data-has-label="true"
                data-slot="base"
              >
                <div
                  aria-hidden="true"
                  data-a11y-ignore="aria-hidden-focus"
                  data-testid="hidden-select-container"
                  style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                >
                  <input
                    style="font-size: 16px;"
                    tabindex="-1"
                    type="text"
                    value=""
                  />
                  <label>
                    Gender
                    <select
                      name="gender"
                      size="2"
                      tabindex="-1"
                      title=""
                    >
                      <option />
                      <option
                        value="male"
                      >
                        Male
                      </option>
                      <option
                        value="female"
                      >
                        Female
                      </option>
                    </select>
                  </label>
                </div>
                <div
                  class="w-full flex flex-col"
                  data-slot="mainWrapper"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="react-aria-:r1a: react-aria-:r16:"
                    class="relative px-3 gap-3 inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 h-10 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-full bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus:bg-gray-50 dark:focus:bg-gray-600 py-[26px] mt-5"
                    data-slot="trigger"
                    id="gender"
                    type="button"
                  >
                    <label
                      class="block absolute subpixel-antialiased text-foreground-500 pointer-events-none will-change-auto origin-top-left rtl:origin-top-right !duration-200 !ease-out transition-[transform,color,left,opacity] motion-reduce:transition-none group-data-[filled=true]:text-foreground z-20 top-1/2 -translate-y-1/2 group-data-[filled=true]:start-0 start-3 group-data-[filled=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden text-xl font-medium pb-1"
                      data-slot="label"
                      id="react-aria-:r16:"
                    >
                      Gender
                    </label>
                    <div
                      class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                      data-slot="innerWrapper"
                    >
                      <span
                        class="text-foreground-500 font-normal w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground"
                        data-slot="value"
                        id="react-aria-:r1a:"
                      >
                         
                      </span>
                    </div>
                    <svg
                      aria-hidden="true"
                      class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                      data-slot="selectorIcon"
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="presentation"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      width="1em"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-3 pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 opacity-50 px-[30px] pt-[15px] pb-[14px] rounded-[10px] border-[1px] h-auto min-w-max w-full mt-8 text-xl font-medium cursor-pointer"
              data-disabled="true"
              disabled=""
              type="submit"
            >
              <div
                class="flex items-center"
              >
                Add Customer
              </div>
            </button>
          </form>
        </div>
      </nav>
      <label
        class="EZDrawer__overlay "
        for="EZDrawer__checkbox"
        id="EZDrawer__overlay"
        style="background-color: rgb(0, 0, 0); opacity: 0.4; z-index: 100;"
      />
    </div>
  </div>
</div>
`;
